### ISSUE TITLE
Cannot download documentation offline

### ISSUE THREAD

**Issue by MagicLegend** [freeCodeCamp/devdocs#1895]
# Bug report

When trying to download documentation for offline use, it will just jump to `error`. After flushing the browser caches it's also causing dynamic pages to not load anymore, which breaks the app.

## OS information

MacOS Ventura (13.0.1), Firefox 108.0.1

## Steps to reproduce

Go to https://devdocs.io/offline and try to install docs.

Note that in Chrome I cannot reproduce this issue, it might be isolated to Firefox.

## More resources

![image](https://user-images.githubusercontent.com/3169104/208849350-65019cca-75ff-4cea-8041-8685b63c4b25.png)

![image](https://user-images.githubusercontent.com/3169104/208851471-f35fa8ec-2f9c-4e54-a931-4b9d2af47c57.png)

**Comment by ojeytonwilliams** [comment]
Hi @MagicLegend I was unable to replicate this on Firefox 108.0.1 (Ubuntu 22.04.1 LTS). Is this still an issue for you?

**Comment by MagicLegend** [comment]
Hi! No, it appears some config changed and it's all good now :)

**Comment by ojeytonwilliams** [comment]
Awesome, thanks for confirming.

**Comment by MagicLegend** [comment]
Hey @ojeytonwilliams,

I'm having this issue again I'm afraid :)

**Comment by Congee** [comment]
How can I help? Both firefox and chrome now default to strict-origin-when-cross-origin regarding the referrer policy. But, only firefox hits this issue.

Any pointers to debug it?

**Comment by Congee** [comment]
seems a dup of https://github.com/freeCodeCamp/devdocs/issues/1417

**Comment by MagicLegend** [comment]
@Congee Is it still a dup if it's still an issue? 🤷 I can confirm I still have a bunch of CORS issues on the site.

**Comment by Congee** [comment]
> 

yes, still an issue for me, too. it's so sad some sites only work in Chrome.

**Comment by jody-frankowski** [comment]
In certain conditions, Firefox will issue a CORS preflight request (AFAICT Chrome never does on devdocs.io). However, these requests fail because CORS isn't activated for your bucket:
```
$ curl -X OPTIONS 'https://documents.devdocs.io/http/db.json?1668720049' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:136.0) Gecko/20100101 Firefox/136.0' -H 'Accept: application/json' -H 'Accept-Language: en_us' -H 'Accept-Encoding: gzip, deflate, br, zstd' -H 'Origin: https://devdocs.io' -H 'Connection: keep-alive' -H 'Referer: https://devdocs.io/' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-site' -H 'Priority: u=0' -H 'Pragma: no-cache' -H 'Cache-Control: no-cache' --compressed
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessForbidden</Code><Message>CORSResponse: CORS is not enabled for this bucket.</Message><Method>OPTIONS</Method><ResourceType>BUCKET</ResourceType><RequestId>GCQFSQETPJEBXMQS</RequestId><HostId>Uj/0hWbwZt/ctI+YoV0g69gzpnRAn/6ajmAI1spOTyy6aYB7Ivb7m/1j62oDEH75hu44XxzmYd0=</HostId></Error>
```

Follow these steps to check for yourself:
- Go to [devdocs.io/offline](https://devdocs.io/offline)
- Open the [Browser Console](https://firefox-source-docs.mozilla.org/devtools-user/browser_console/) (`Ctrl-Shift-J`/`Cmd-Shift-J`)
- Open the network console
- Force a full refresh with `Ctrl-Shift-R`/`Cmd-Shift-J`
- Install one of the offline doc
- In the network console, you should see a successful `GET ...db.json` request
- Do a simple refresh with `Ctrl-R`/`Cmd-R`
- Install one of the offline doc
- In the browser console, you should see an unsuccessful `GET ...db.json` request and its unsuccessful `OPTIONS` preflight request

I don't know why Firefox issues the preflight request in this specific case and if chrome should but doesn't.
